core/devmode/src/main/java/io/quarkus/dev/DevModeMain.java

```diff
@@ -110,6 +114,36 @@ public void start() throws Exception {
         }
     }
 
+    // copies the .env file to the directory where the process is running
+    // this is done because for the .env file to take effect, it needs to be
+    // in the same directory as the running process
+    private void copyDotEnvFile() {
+        File projectDir = context.getProjectDir();
+        if (projectDir == null) { // this is the case for QuarkusDevModeTest
+            return;
+        }
+        Path currentDir = Paths.get("").toAbsolutePath().normalize();
+        Path dotEnvPath = projectDir.toPath().resolve(".env");
+        if (Files.exists(dotEnvPath)) {
+            try {
+                Path link = currentDir.resolve(".env");
+                silentDeleteFile(link);
+                // create a symlink to ensure that user updates to the file have the expected effect in dev-mode
+                Files.createSymbolicLink(link, dotEnvPath);
```

Comment:
This seems wrong... unless we know for a fact that the current directory is going to be `target/`?  We don't want to modify anything outside `target` for any reason.  It would be better to modify the config source impl if that's the case.

Reply:
The reason I did this is because as a user I expect that if my `${rootDir}/.env` file is going to be picked up when I do `mvn quarkus:dev`, then I would also expect it to be monitored for changes. With this way of doing things, those changes are picked up.

It doesn't really have to do with `./target` at all, unless I am missing something. 

WDYT?

Reply:
The question is, where is this symbolic link being created, in practical terms?

Reply:
Inside `target`. Which is why I mentioned that updates work, but deleting the original `.env` doesn't (although I wouldn't expect the user to do that)

Reply:
OK, that's what I wanted to know.  Thanks!

Reply:
:+1: